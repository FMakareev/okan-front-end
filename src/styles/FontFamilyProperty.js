/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
