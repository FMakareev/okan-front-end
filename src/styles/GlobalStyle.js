import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
