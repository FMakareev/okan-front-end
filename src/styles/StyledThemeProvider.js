import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
