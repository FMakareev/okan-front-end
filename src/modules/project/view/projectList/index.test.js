import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider as ProviderRedux } from 'react-redux';

import { BrowserRouter } from 'react-router-dom';
import { ProjectListPage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';

const store = createStore(() => {});

test('ProjectListPage: рендер без ошибок', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={store}>
        <BrowserRouter>
          <ProjectListPage />
        </BrowserRouter>
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});
