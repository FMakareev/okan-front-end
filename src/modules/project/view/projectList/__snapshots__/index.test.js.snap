// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListPage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb gPUjkv"
>
  <div>
    <div
      className="sc-1jxwge6-0 dgupjs-0 sc-1kgy4mw-0 iiLzzb"
    >
      <i
        className="sc-1kgy4mw-1 dQvUyw"
        color="color6"
        fill="currentColor"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <svg
          fill="currentColor"
          height={64}
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 24 24"
          width={64}
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </i>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 egWvTL"
        color="color11"
        fontSize={7}
        fontWeight="bold"
      >
        Error
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 iyPpaZ"
        color="#000"
        fontSize={5}
      >
        unexpected error
      </div>
    </div>
  </div>
</div>
`;
